cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/WorkflowScheduleUtils.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package io.cdap.cdap.datapipeline.service;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.schedule.ProgramStatusTriggerInfo;
+import io.cdap.cdap.etl.api.lineage.field.FieldOperation;
+import io.cdap.cdap.etl.common.BasicArguments;
+import io.cdap.cdap.etl.common.FieldOperationTypeAdapter;
+import io.cdap.cdap.etl.proto.v2.ArgumentMapping;
+import io.cdap.cdap.etl.proto.v2.PluginPropertyMapping;
+import io.cdap.cdap.etl.proto.v2.TriggeringPipelineId;
+import io.cdap.cdap.etl.proto.v2.TriggeringPropertyMapping;
+import io.cdap.cdap.internal.io.SchemaTypeAdapter;
+import org.slf4j.Logger;
+
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+
+public final class WorkflowScheduleUtils {
+  private static final Gson GSON = new GsonBuilder()
+    .registerTypeAdapter(Schema.class, new SchemaTypeAdapter())
+    .registerTypeAdapter(FieldOperation.class, new FieldOperationTypeAdapter()).create();
+  private static final String RESOLVED_PLUGIN_PROPERTIES_MAP = "resolved.plugin.properties.map";
+  private static final Type STAGE_PROPERTIES_MAP = new TypeToken<Map<String, Map<String, String>>>() { }.getType();
+
+  private WorkflowScheduleUtils() {
+  }
+
+  /**
+   * Parse the arguments mapping from the pipeline schedule.
```

Comment:
this description should have more detail. It should include some example input and what gets added to the mappings.

Somebody should be able to read this and tell what the method does, especially since the code isn't the clearest.